UnityProject/Assets/Scripts/Items/Weapons/Explosive.cs

```diff
@@ -85,17 +88,35 @@ private void Detonate()
 			if (isServer)
 			{
 				// Get data before despawning
-				var explosionMatrix = registerItem.Matrix;
 				var worldPos = objectBehaviour.AssumedWorldPositionServer();
 
 				// Despawn the explosive
 				_ = Despawn.ServerSingle(gameObject);
+				Explosion.StartExplosion(worldPos, explosiveStrength);
+			}
+
+		}
 
-				// Explosion here
-				var explosionGO = Instantiate(explosionPrefab, explosionMatrix.transform);
-				explosionGO.transform.position = worldPos;
-				explosionGO.Explode(explosionMatrix);
+		[Server]
+		private void AttachExplosive(GameObject target, Vector2 targetPostion)
+		{
+			if (target.TryGetComponent<PushPull>(out var handler))
+			{
+				Inventory.ServerDrop(pickupable.ItemSlot, targetPostion);
+				attachedToObject = target;
+				UpdateManager.Add(CallbackType.UPDATE,UpdateBombPosition);
+				if (spriteHandler != null) spriteHandler.transform.localScale = new Vector3(0.6f, 0.6f, 0.6f);
```

Comment:
Don't think scale is sync'd.

Not even sure we have a component that does it already, you're gonna have to make a new component to do it using a sync var.

Reply:
if i turn the scale thing into a command function called by the server; would it work on both the client and server? or will it try kicking the player again?

Reply:
> if i turn the scale thing into a command function called by the server; would it work on both the client and server? or will it try kicking the player again?

Will need to be a sync var for late joiners. And make a new script for it as it could be useful in the future, call it NetworkScale or something like that

Reply:
Also might be better to make it a periodic update on like `0.1f` as it doesnt need to be every frame